src/closed.jl

```diff
@@ -0,0 +1,54 @@
+"""
+Defines the ClosedSet type.
+
+A ClosedSet(left, right) is an interval set where both bounds are closed i.e. the range is [left, right]
```

Comment:
Simon makes a very good point. Formally speaking, a "closed set" is an *arbitrary* set that contains its limit points whereas a closed interval is much more specific (and is what is implemented).